src/gui/qgsabstractdatasourcewidget.h

```diff
@@ -50,7 +50,7 @@ class GUI_EXPORT QgsAbstractDataSourceWidget : public QDialog
      * Store a pointer to the map canvas to retrieve extent and CRS
      * Used to select an appropriate CRS and possibly to retrieve data only in the current extent
      */
-    void setMapCanvas( const QgsMapCanvas *mapCanvas );
+    void setMapCanvas( QgsMapCanvas *mapCanvas );
```

Comment:
This still needs to be made virtual to avoid the need for the mapCanvasChanged signal